sklearn/kernel_ridge/kernel_ridge.py

```diff
@@ -143,23 +114,42 @@ def fit(self, X, y=None, sample_weight=None):
         self : returns an instance of self.
         """
         # Convert data
-        X, y = check_X_y(X, y, accept_sparse=("csr", "csc"), multi_output=True,
-                         y_numeric=True)
         if sample_weight is not None and not isinstance(sample_weight, float):
             sample_weight = check_array(sample_weight, ensure_2d=False)
 
         K = self._get_kernel(X)
         alpha = np.atleast_1d(self.alpha)
 
+        copy = self.kernel == "precomputed"
+
         ravel = False
-        if len(y.shape) == 1:
-            y = y.reshape(-1, 1)
-            ravel = True
+        if self._estimator_type == "regressor":
+            X, y_ = check_X_y(X, y, accept_sparse=("csr", "csc"),
+                              multi_output=True, y_numeric=True)
+            if len(y_.shape) == 1:
+                y_ = y_.reshape(-1, 1)
+                ravel = True
+        elif self._estimator_type == "classifier":  # Multiclass classification
+            X, y = check_X_y(X, y)
```

Comment:
Why do we not accept sparse?

Reply:
I didn't understand them pretty well, but I'm going to change the `check_X_y`. Should I prepare a test with sparse data, or common tests check it? Is there a list of what common tests check and what not?

Reply:
Done